Source/Core/DolphinQt/FIFO/FIFOAnalyzer.cpp

```diff
@@ -550,6 +597,51 @@ void FIFOAnalyzer::UpdateDescription()
     else
       text += QString::fromStdString(desc);
   }
+  else if ((*cmddata & 0xC0) == 0x80)
+  {
+    const auto name = GetPrimitiveName(*cmddata);
+    const u16 vertex_count = Common::swap16(cmddata + 1);
+
+    ASSERT(entry_size > 3);
+    const u32 primitive_size = entry_size - 3;
+    const u8* primitive_data = cmddata + 3;
+    const u8* const primitive_end = primitive_data + primitive_size;
+
+    text = tr("Primitive ");
+    text += QString::fromStdString(name);
+    text += QLatin1Char{'\n'};
+
+    u32 vertex_size;
+    if (vertex_count != 0)
+    {
+      if ((primitive_size % vertex_count) == 0)
+      {
+        vertex_size = primitive_size / vertex_count;
+      }
+      else
+      {
+        text += tr("NOTE: Stream size doesn't match actual data length");
```

Comment:
Just want to say, I saw this quite a bit in one of my fifo logs.  Not sure if that's valid or not?

Reply:
That message existed beforehand (and should occur with the same frequency as before).  I've taken a brief look and it seems like sometimes there can be multiple draw commands in a row that get bundled into one object (but then the size doesn't match), and there also sometimes is NOP padding at the end (which will also cause the size not to match).  It's something that could be handled much more elegantly if the vertex format were known at that point in the code, but it's not known.  (I still need to look at what code is responsible for breaking the FIFO into objects and how it can be tweaked to give better results/how it handles these things.)

Reply:
I've since changed it so that the object size is known; this issue no longer happens.  Since the vertex size is known, the number of bytes each draw command uses also is known, so multiple draw commands in a row are also handled better.  (One other fun edge case is that you can have a draw command that draws 0 vertices... which the code previously wasn't handling in any sane way, because it couldn't even do this size check in that case.  Now it's just treated the exact same as the other cases.)